tests/test_mysql.py

```diff
@@ -1,52 +1,149 @@
-import time
+
+# project
+from tests.common import AgentCheckTest
+
 
 @attr(requires='mysql')
-class TestMySql(unittest.TestCase):
+class TestMySql(AgentCheckTest):
+    CHECK_NAME = 'mysql'
+
+    METRIC_TAGS = ['tag1', 'tag2']
+    SC_TAGS = ['host:localhost', 'port:0']
+
+    MYSQL_CONFIG = {
+        'init_config': {},
+        'instances': [{
+            'server': 'localhost',
+            'user': 'dog',
+            'pass': 'dog',
+            'options': {'replication': True},
+            'tags': METRIC_TAGS
+        }]
+    }
+
+    FAILURE_CONNECTION = {
+        'init_config': {},
+        'instances': [{
+            'server': 'localhost',
+            'user': 'unknown',
+            'pass': 'dog',
+        }]
+    }
+
+    # Available by default on MySQL > 5.5
+    INNODB_METRICS = [
+        "mysql.innodb.buffer_pool_free",
+        "mysql.innodb.buffer_pool_used",
+        "mysql.innodb.buffer_pool_total",
+        "mysql.innodb.buffer_pool_utilization"
+    ]
+
+    REPLICATION_METRICS = [
+        "mysql.replication.slave_running"
+    ]
+
+    KEY_CACHE = [
+        "mysql.performance.key_cache_utilization"
+    ]
+
+    # Available on Linux
+    SYSTEM_METRICS = [
+        "mysql.performance.user_time",
+        "mysql.performance.kernel_time"
+    ]
+
+    COMMON_GAUGES = [
+        'mysql.net.max_connections',
+        'mysql.performance.open_files',
+        'mysql.performance.table_locks_waited',
+        'mysql.performance.threads_connected',
+        'mysql.innodb.current_row_locks',
+        'mysql.performance.open_tables',
+    ]
+
+    COMMON_RATES = [
+        'mysql.net.connections',
+        'mysql.innodb.data_reads',
+        'mysql.innodb.data_writes',
+        'mysql.innodb.os_log_fsyncs',
+        'mysql.innodb.buffer_pool_size',
+        'mysql.performance.slow_queries',
+        'mysql.performance.questions',
+        'mysql.performance.queries',
+        'mysql.performance.com_select',
+        'mysql.performance.com_insert',
+        'mysql.performance.com_update',
+        'mysql.performance.com_delete',
+        'mysql.performance.com_insert_select',
+        'mysql.performance.com_update_multi',
+        'mysql.performance.com_delete_multi',
+        'mysql.performance.com_replace_select',
+        'mysql.performance.qcache_hits',
+        'mysql.innodb.mutex_spin_waits',
+        'mysql.innodb.mutex_spin_rounds',
+        'mysql.innodb.mutex_os_waits',
+        'mysql.performance.created_tmp_tables',
+        'mysql.performance.created_tmp_disk_tables',
+        'mysql.performance.created_tmp_files',
+        'mysql.innodb.row_lock_waits',
+        'mysql.innodb.row_lock_time',
+    ]
+
     def setUp(self):
-        # This should run on pre-2.7 python so no skiptest
+        """
+        Skip tests on Python < 2.7
+        """
         self.skip = False
         try:
-            import pymysql
+            import pymysql  # noqa
         except ImportError:
             self.skip = True
```

Comment:
can you use nose `SkipTest` with a string explaining that like `pymsql not runinng on python < 2.7`
also add something to skip the CI more upstream (otherwise we'll still install and set up mysql in the CI to eventually skip the test suite).